packages/nuxt/test/runtime/utils/route-extraction.test.ts

```diff
@@ -0,0 +1,272 @@
+import { describe, expect, it } from 'vitest';
+import type { NuxtPageSubset } from '../../../src/runtime/utils/route-extraction';
+import { extractParametrizedRouteFromContext } from '../../../src/runtime/utils/route-extraction';
+
+/** Creates a mock NuxtPage object with all existing pages. Nuxt provides this during the build time in the "pages:extend" hook.
+ *  The content is inspired by a real-world example. */
+const createMockPagesData = (
+  overrides: NuxtPageSubset[] = [],
+  addDefaultPageData: boolean = true,
+): NuxtPageSubset[] => {
+  const defaultBase = [
+    // Basic routes
+    { path: '/', file: '/private/folders/application/pages/index.vue' },
+    { path: '/simple-page', file: '/private/folders/application/pages/simple-page.vue' },
+    { path: '/a/nested/simple-page', file: '/private/folders/application/pages/a/nested/simple-page.vue' },
+    // Dynamic routes (directory and file)
+    { path: '/user/:userId()', file: '/private/folders/application/pages/user/[userId].vue' },
+    { path: '/group-:name()/:id()', file: '/private/folders/application/pages/group-[name]/[id].vue' },
+    // Catch-all routes
+    { path: '/catch-all/:path(.*)*', file: '/private/folders/application/pages/catch-all/[...path].vue' },
+  ];
+
+  return [...(addDefaultPageData ? defaultBase : []), ...overrides];
+};
+
+// The base of modules when loading a specific page during runtime (inspired by real-world examples).
+const defaultSSRContextModules = new Set([
+  'node_modules/nuxt/dist/app/components/nuxt-root.vue',
+  'app.vue',
+  'components/Button.vue',
+  // ...the specific requested page is added in the test (e.g. 'pages/user/[userId].vue')
+]);
```

Comment:
I will highlight the new stuff for the tests in this PR to make it easier to look through.

Those are the new function to create some default data.